draco/analysis/delay.py

```diff
@@ -54,10 +83,31 @@ def process(self, ss):
         ssv = ss.vis[:].view(np.ndarray)
         ssw = ss.weight[:].view(np.ndarray)
 
+        ubase, uinv = np.unique(
+            tel.baselines[:, 0] + 1.0J * tel.baselines[:, 1],
+            return_inverse=True
+        )
+        ubase = ubase.view(np.float64).reshape(-1, 2)
+
         for lbi, bi in ss.vis[:].enumerate(axis=1):
 
-            freq_weight = np.median(ssw[:, lbi], axis=1)
-            NF = null_delay_filter(freq, self.delay_cut, freq_weight)
+            # Select the baseline length to use
+            baseline = ubase[uinv[bi]]
+            if self.telescope_orientation == 'NS':
+                baseline = abs(baseline[1])  # Y baseline
+            elif self.telescope_orientation == 'EW':
+                baseline = abs(baseline[0])  # X baseline
+            else:
+                baseline = np.linalg.norm(baseline)  # Norm
```

Comment:
I think this collapses both axes and thus is flat with baselines. Try `np.hypot` instead, or specify the `axis` argument.

Reply:
I am inside of a baseline loop here, so I think I selected a single baseline entry right before the `if` statement. The only axis remaining is the xy one.
I couldn't test though because I can't get any jobs running on cedar.

Reply:
Testing now.

Reply:
Oh. Good point. My bad!